src/Illuminate/Collections/functions.php

```diff
@@ -2,6 +2,47 @@
 
 namespace Illuminate\Support;
 
+use Illuminate\Database\Eloquent\Model;
+
+if (! function_exists('Illuminate\Support\is_model')) {
+    /**
+     * Determine if the given value is a model.
+     *
+     * @internal
+     *
+     * @template TModel
+     *
+     * @param  TModel  $model
+     * @return bool
+     */
+    function is_model($model)
+    {
+        return $model instanceof Model;
+    }
+}
+
+if (! function_exists('Illuminate\Support\model_key')) {
+    /**
+     * Return the key for the given model.
+     *
+     * @internal
+     *
+     * @template TModel
+     *
+     * @param  TModel  $model
+     * @param  callable(TModel): mixed  $callback
+     * @return mixed
+     */
+    function model_key($model, $callback = null)
+    {
+        if (is_model($model)) {
+            return $model->getKey();
+        }
+
+        return with($model, $callback);
```

Comment:
What's the benefit of returning _something_ if it's not a model?

Reply:
I can understand that one might like to have a shortcut in that situation, yet, when only reading the latter, this is not intuitive and doesn't make a lot of sense semantically.